Source/WebKit/UIProcess/API/Cocoa/APIAttachmentCocoa.mm

```diff
@@ -124,61 +124,69 @@ static BOOL isDeclaredOrDynamicTypeIdentifier(NSString *type)
 
 std::optional<uint64_t> Attachment::fileSizeForDisplay() const
 {
-    auto fileWrapper = this->fileWrapper();
+    Locker locker { m_fileWrapperLock };
 
-    if (![fileWrapper isRegularFile]) {
+    if (![m_fileWrapper isRegularFile]) {
         // FIXME: We should display a size estimate for directory-type file wrappers.
         return std::nullopt;
     }
 
-    if (auto fileSize = [[fileWrapper fileAttributes][NSFileSize] unsignedLongLongValue])
+    if (auto fileSize = [[m_fileWrapper fileAttributes][NSFileSize] unsignedLongLongValue])
         return fileSize;
 
-    return [fileWrapper regularFileContents].length;
+    return [m_fileWrapper regularFileContents].length;
 }
 
 RefPtr<WebCore::FragmentedSharedBuffer> Attachment::enclosingImageData() const
 {
     if (!m_hasEnclosingImage)
         return nullptr;
 
-    auto fileWrapper = this->fileWrapper();
+    NSData *data = nil;
+    {
+        Locker locker { m_fileWrapperLock };
 
-    if (![fileWrapper isRegularFile])
-        return nullptr;
+        if (![m_fileWrapper isRegularFile])
+            return nullptr;
 
-    NSData *data = [fileWrapper regularFileContents];
-    if (!data)
-        return nullptr;
+        data = [m_fileWrapper regularFileContents];
+        if (!data)
```

Comment:
Nit: this check could be outside the lock scope.

Reply:
Fixed!